### PR TITLE
Doma-6526 pdf receipt for residents

### PR DESCRIPTION
**Pull request by dkoviazin**

### CODE REVIEW

File: apps/condo/domains/billing/schema/BillingReceiptFile.js [hunk 1]
```diff
@@ -2,30 +2,77 @@
...
+            graphQLReturnType: 'File',
+            resolver: async (item, _, { authedItem }) => {
+                // We are changing link to publicData only for not verified residents. In other cases we return sensitive data files
+                let file = item.sensitiveDataFile
+                if (authedItem.type === RESIDENT) {
+                    // Resident already has an access to billing receipt, so we only need to check if he is approved
+                    const { account } = await getById('BillingReceipt', item.receipt)
+                    const isApproved = await isResidentVerified(authedItem, account)
+                    if (!isApproved) {
+                        file = item.publicDataFile
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/BillingReceiptFile.js:48]: It's better not to use sensitive data as default value
```suggestion
                let file = item.publicDataFile
                if (authedItem.type === RESIDENT) {
                    // Resident already has an access to billing receipt, so we only need to check if he is approved
                    const { account } = await getById('BillingReceipt', item.receipt)
                    const isApproved = await isResidentVerified(authedItem, account)
                    if (isApproved) {
                        file = item.sensitiveDataFile
```

File: apps/condo/domains/billing/schema/BillingReceiptFile.js [hunk 2]
```diff
@@ -68,8 +115,19 @@ const BillingReceiptFile = new GQLListSchema('BillingReceiptFile', {
         auth: true,
     },
     hooks: {
-        // Set meta to file for checking permissions { listkey, id }
-        afterChange: fileMetaAfterChange,
+        afterChange: async ({ updatedItem, listKey }) => {
+            if (updatedItem && Adapter.acl && Adapter.acl.setMeta) {
+                const sensitiveFile = get(updatedItem, 'sensitiveDataFile.filename')
+                const publicFile = get(updatedItem, 'publicDataFile.filename')
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/BillingReceiptFile.js:121]: Lodash always casts a string path to an array. So we can decrease the number of calculations.
```suggestion
                const sensitiveFile = get(updatedItem, ['sensitiveDataFile', 'filename'])
                const publicFile = get(updatedItem, ['publicDataFile', 'filename'])
```

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/billing/schema/BillingReceiptFile.js (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/schema/BillingReceiptFile.js:121]: it's hard to use such syntax on constructions like
```
const residentName = get(residents, '[0].name')
```
So, I prefer string variant

File: apps/condo/domains/billing/schema/BillingReceiptFile.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/schema/BillingReceiptFile.js]: Done

**Pull request merged**